compiler/expressions.go

```diff
@@ -969,7 +981,12 @@ func (fc *funcContext) translateBuiltin(name string, sig *types.Signature, args
 	case "delete":
 		args = fc.expandTupleArgs(args)
 		keyType := fc.pkgCtx.TypeOf(args[0]).Underlying().(*types.Map).Key()
-		return fc.formatExpr(`delete %e[%s.keyFor(%s)]`, args[0], fc.typeName(keyType), fc.translateImplicitConversion(args[1], keyType))
+		return fc.formatExpr(
+			`(typeof %1e.delete === "function" && %1e.delete(%2s.keyFor(%3s)))`,
```

Comment:
Similar to above, introducing a `$mapDelete()` helper should make the generated code more compact and compressible.